tools/transformer/src/transformations/oneOf.js

```diff
@@ -121,6 +129,24 @@ function transformOneOfProperties(parentObject, api) {
       ...parentObject.properties,
       ...childProperties,
     };
+
+    // Collect required fields from the child
+    if (childObject.required) {
```

Comment:
This logic should give you some context on complexity of that transformation.
Results seem off - will likely need more investigation.